pkg/virt-launcher/virtwrap/converter/pci-placement.go

```diff
@@ -48,32 +52,45 @@ func PlacePCIDevicesOnRootComplex(spec *api.DomainSpec) (err error) {
 		if input.Bus != v1.VirtIO {
 			continue
 		}
-		spec.Devices.Inputs[i].Address, err = assigner.PlacePCIDeviceAtNextSlot(input.Address)
+		spec.Devices.Inputs[i].Address, err = fn(input.Address)
 		if err != nil {
 			return err
 		}
 	}
 	for i, watchdog := range spec.Devices.Watchdogs {
-		spec.Devices.Watchdogs[i].Address, err = assigner.PlacePCIDeviceAtNextSlot(watchdog.Address)
+		spec.Devices.Watchdogs[i].Address, err = fn(watchdog.Address)
 		if err != nil {
 			return err
 		}
 	}
 	if spec.Devices.Rng != nil {
-		spec.Devices.Rng.Address, err = assigner.PlacePCIDeviceAtNextSlot(spec.Devices.Rng.Address)
+		spec.Devices.Rng.Address, err = fn(spec.Devices.Rng.Address)
 		if err != nil {
 			return err
 		}
 	}
 	if spec.Devices.Ballooning != nil {
-		spec.Devices.Ballooning.Address, err = assigner.PlacePCIDeviceAtNextSlot(spec.Devices.Ballooning.Address)
+		spec.Devices.Ballooning.Address, err = fn(spec.Devices.Ballooning.Address)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func CountVirtIOPortsUsed(spec *api.DomainSpec) (count int, err error) {
+	err = iteratePCIAddresses(spec, func(address *api.Address) (*api.Address, error) {
```

Comment:
I'm unsure whether the function counts all used PCI slots or just VIRTIO PCI slots because there seems to be nothing VIRTIO-specific here?

Reply:
Yeah true bad name nothing virtio specific there

Reply:
The caller expects this to count all PCI devices, not merely virtio devices, so it is merely a bad method naming choice.